pipeline/src/main/java/com/iluwatar/pipeline/App.java

```diff
@@ -0,0 +1,116 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.pipeline;
+
+/**
+ * The Pipeline pattern uses parallel tasks and concurrent queues to process a
+ * sequence of input values. Each task implements a stage of the pipeline, and 
+ * the queues act as buffers that allow the stages of the pipeline to execute 
+ * concurrently, even though the values are processed in order. In this program,
+ * sentences are decoded in four stages, where, each stage is dedicated to just
+ * one single task.
+ */
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ *
+ * @author Ayush
+ */
+public class App {
+  
+  public static Queue buffer1;
+  public static Queue buffer2;
+  public static Queue buffer3;
+  public static Queue buffer4;
+    /**
+     * Program entry point
+     */
+  public static void main(String[] args) {
+    Queue data = Data.data();
+    System.out.println("Data received");
+    
+    Filter stage1 = new Stage1();
+    Filter stage2 = new Stage2();
+    Process stage3 = new Stage3();
+    Print stage4 = new Stage4();
+    
+    buffer1 = new LinkedList();
+    buffer2 = new LinkedList();
+    buffer3 = new LinkedList();
+    buffer4 = new LinkedList();
+    System.out.println("Buffers initialised");
+    
+    int time = 1;
+    
+    /**
+     * Each iteration of the loop specify a single instance where each process
+     * is carried out simultaneously.
+     */ 
+
+    while (!(data.isEmpty())) {
+        
+      System.out.println("\nTime " + time + " : ");
+      String newData = (String) data.remove();
+      buffer1.add(newData);
+      if (!buffer4.isEmpty()) {
+        stage4.printData();
+      }
+      if (!buffer3.isEmpty()) {
+        stage3.process();
+        System.out.println("Data processed through 3rd stage");
+      }
+      if (!buffer2.isEmpty()) {
+        stage2.removeChar();
+        System.out.println("Data filtered through 2nd stage");
+      }
+      if (!buffer1.isEmpty()) {
+        stage1.removeChar();
+        System.out.println("Data filtered through 1st stage");
+      }
+      time++;
+    }
+    
+    while (!buffer4.isEmpty()) {
+        
+      System.out.println("\nTime " + time + " : ");
+      stage4.printData();  
+      if (!buffer3.isEmpty()) {
+        stage3.process();
+        System.out.println("Data processed through 3rd stage");
+      }
+      if (!buffer2.isEmpty()) {
+        stage2.removeChar();
+        System.out.println("Data filtered through 2nd stage");
+      }
+      if (!buffer1.isEmpty()) {
+        stage1.removeChar();
+        System.out.println("Data filtered through 1st stage");
+      }
+      time++;
+    }
+    
+  }

```

Comment:
We should get rid of the procedural processing loop. I'm thinking that simple `pipeline.start()` or similar should run the whole pipeline processing.